python/python/tests/tests_server_modules/test_json.py

```diff
@@ -359,3 +359,87 @@ async def test_json_clear(self, glide_client: TGlideClient):
+        result = await json.numincrby(glide_client, key, "$.key1", 5)
+        assert result == b"[6]"  # Expect 1 + 5 = 6
+
+        # Increment float value (key2) by 2.5
+        result = await json.numincrby(glide_client, key, "$.key2", 2.5)
+        assert result == b"[6]"  # Expect 3.5 + 2.5 = 6
+
+        # Increment nested object (key3.nested_key.key1[0]) by 7
+        result = await json.numincrby(glide_client, key, "$.key3.nested_key.key1[1]", 7)
+        assert result == b"[12]"  # 4+7=12
+
+        # Increment array element (key4[1]) by 1
+        result = await json.numincrby(glide_client, key, "$.key4[1]", 1)
+        assert result == b"[3]"  # Expect 2 + 1 = 3
+
+        # Increment zero value (key5) by 10.23 (float number)
+        result = await json.numincrby(glide_client, key, "$.key5", 10.23)
+        assert result == b"[10.23]"  # Expect 0 + 10.23 = 10.23
+
+        # Increment a string value (key6) by a number
+        result = await json.numincrby(glide_client, key, "$.key6", 99)
+        assert result == b"[null]"  # Expect null
+
+        # Increment a None value (key7) by a number
+        result = await json.numincrby(glide_client, key, "$.key7", 51)
+        assert result == b"[null]"  # Expect null
+
+        # Check increment for all numbers in the document using JSON Path (First Null: key3 as an entire object. Second Null: The path checks under key3, which is an object, for numeric values).
+        result = await json.numincrby(glide_client, key, "$..*", 5)
+        assert (
+            result
+            == b"[11,11,null,null,15.23,null,null,null,null,null,9,17,6,8,8,null,74]"
+        )
+
+        # Check for multiple path match in enhanced
+        result = await json.numincrby(glide_client, key, "$..key1", 1)
+        assert result == b"[12,null,75]"
+
+        # EXAMPLES FOR LEGACY PATH CAN BE FOUND HERE
+        # Increment integer value (key1) by 5 (integer)
+        result = await json.numincrby(glide_client, key, "key1", 5)
+        assert result == b"17"  # Expect 12 + 5 = 17
+
+        # Increment float value (key2) by 2.5 (a float number)
+        result = await json.numincrby(glide_client, key, "key2", 2.5)
+        assert result == b"13.5"  # Expect 11 + 2.5 = 13.5
+
+        # Increment nested value (key3.nested_key.key1[0]) by 7
+        result = await json.numincrby(glide_client, key, "key3.nested_key.key1[0]", 7)
+        assert result == b"16"  # Expect 9 + 7 = 16
+
+        # Increment array element (key4[1]) by 1
+        result = await json.numincrby(glide_client, key, "key4[1]", 1)
+        assert result == b"9"  # Expect 8 + 1 = 9
+
+        # Increment a float value (key5) by 10.2 (a float number)
+        result = await json.numincrby(glide_client, key, "key5", 10.2)
+        assert result == b"25.43"  # Expect 15.23 + 10.2 = 25.43
+
+        # Check for multiple path match in legacy
+        result = await json.numincrby(glide_client, key, "..key1", 1)
+        assert result == b"76"
```

Comment:
can you make sure that the rest of them were updated as well?

Reply:
please make sure that get $..key shows all have been incremented

Reply:
the get command returned only the first value. I did increment all values by 0 and checked $..key

Reply:
use get with $

Reply:
done